packages/transaction-controller/src/gas-flows/OracleLayer1GasFeeFlow.ts

```diff
@@ -32,22 +32,37 @@ const GAS_PRICE_ORACLE_ABI = [
  * Layer 1 gas fee flow that obtains gas fee estimate using an oracle smart contract.
  */
 export abstract class OracleLayer1GasFeeFlow implements Layer1GasFeeFlow {
-  readonly #oracleAddress: Hex;
+  readonly #oracleAddress: Hex | undefined;
 
   readonly #signTransaction: boolean;
 
-  constructor(oracleAddress: Hex, signTransaction?: boolean) {
+  constructor(oracleAddress?: Hex, signTransaction?: boolean) {
```

Comment:
Can we simplify by removing this and relying only on `getOracleAddressForChain` which could return the default vault in this class rather than in the `Optimism` class?